minorminer/utils/zephyr/node_edge.py

```diff
@@ -0,0 +1,823 @@
+                f"shape and ccoord must be both quotient or non-quotient, got {self._shape}, {coord}"
+            )
+        if (self._shape.t is not None) and (coord.k not in range(self._shape.t)):
+            raise ValueError(f"Expected k to be in {range(self._shape.t)}, got {coord.k}")
+
+        # check x, y value of coord is consistent with m
+        if self._shape.m is not None:
+            if not all(val in range(4 * self._shape.m + 1) for val in (coord.x, coord.y)):
+                raise ValueError(
+                    f"Expected ccoord.x and ccoord.y to be in {range(4*self._shape.m+1)}, got {coord.x, coord.y}"
+                )
+
+    def _set_ccoord(self, coord: CartesianCoord | tuple[int], check_node_valid: bool) -> CartesianCoord:
+        """Returns the :class:`CartesianCoord` corresponding to ``coord``.
+
+        Args:
+            coord (CartesianCoord or tuple[int]): The Cartesian coordinate to set.
+        """
+        if isinstance(coord, tuple):
+            coord = CartesianCoord(*coord)
+
+        for c in coord:
+            if (c is not None) and (not isinstance(c, int)):
+                raise TypeError(f"Expected elements of ccoord to be 'int' or None, got {type(c)}")
+
+        if check_node_valid:
+            self._check_ccoord_val(coord)
+            self._check_ccoord_shape(coord)
+        
+        return coord
+
+    @staticmethod
+    def tuple_to_coord(coord: tuple[int]) -> CartesianCoord | ZephyrCoord:
+        """Takes a tuple[int] and returns the corresponding ``CartesianCoord`` or ``ZephyrCoord``"""
+        if (not isinstance(coord, tuple)) or (not all(isinstance(c, int) for c in coord)):
+            raise TypeError(f"Expected {coord} to be a tuple[int], got {coord}")
+
+        if any(c < 0 for c in coord):
+            raise ValueError(f"Expected elements of coord to be non-negative, got {coord}")
+
+        len_coord = len(coord)
+        if len_coord in (2, 3):
+            x, y, *k = coord
+            if x % 2 == y % 2:
+                raise ValueError(f"Expected x, y to differ in parity, got {x, y}")
+
+            return CartesianCoord(x=x, y=y) if len(k) == 0 else CartesianCoord(x=x, y=y, k=k[0])
+
+        if len_coord in (4, 5):
+            u, w, *k, j, z = coord
+            for var, val in [("u", u), ("j", j)]:
+                if not val in [0, 1]:
+                    raise ValueError(f"Expected {var} to be in [0, 1], got {val}")
+
+            if len(k) == 0:
+                return ZephyrCoord(u=u, w=w, j=j, z=z)
+
+            return ZephyrCoord(u=u, w=w, k=k[0], j=j, z=z)
+            
+        raise ValueError(f"coord can have length 2, 3, 4 or 5, got {len_coord}")
+
+    @property
+    def zcoord(self) -> ZephyrCoord:
+        """Returns :class:`ZephyrCoord` corresponding to ccoord."""
+        return cartesian_to_zephyr(self._ccoord)
+
+    @property
+    def node_kind(self) -> NodeKind:
+        """Returns the :class:`NodeKind` of ``self``."""
```

Comment:
Two guidelines that apply not just to this property:
- [property docstring](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) should be consistent with an attribute docstring
- refer to type, instead "self"
```suggestion
        """The :class:`NodeKind` of the ``ZNode``."""
```